test/unit/spa.test.js

```diff
@@ -1,161 +1,174 @@
 import consola from 'consola'
 import { loadFixture, getPort, Nuxt, wChunk } from '../utils'
 
-let nuxt, port
-const url = route => 'http://localhost:' + port + route
-
-const renderRoute = async (_url) => {
-  const window = await nuxt.server.renderAndGetWindow(url(_url))
-  const head = window.document.head.innerHTML
-  const html = window.document.body.innerHTML
-  return { window, head, html }
-}
+// Runs tests in specified router mode (either "hash" or "history").
+function spaTests ({ isHashMode }) {
```

Comment:
Can we extract the hash mode specific test cases into a separted file like `spa.hash.test.js` for more readable and easier to maintain in future ?

Reply:
There are no "hash specific" testcases per se. All are valid testcases for both modes (apart from one test which fails in hash mode for some unknown reason). So it wouldn't make sense to create two identical sets of tests and two identical sets of files in a new fixture.